src/common/cluster-store.ts

```diff
@@ -229,14 +267,37 @@ export class ClusterStore extends BaseStore<ClusterStoreModel> {
     const currentClusters = this.clusters.toJS();
     const newClusters = new Map<ClusterId, Cluster>();
     const removedClusters = new Map<ClusterId, Cluster>();
+    const deadClusters = new Map<ClusterId, [ClusterModel, LoadKubeError]>();
 
     // update new clusters
     for (const clusterModel of clusters) {
       let cluster = currentClusters.get(clusterModel.id);
       if (cluster) {
         cluster.updateModel(clusterModel);
       } else {
-        cluster = new Cluster(clusterModel);
+        try {
+          cluster = new Cluster(clusterModel);
+        } catch (err) {
+          if (err instanceof LoadKubeError) {
+            deadClusters.set(clusterModel.id, [clusterModel, err]);
+            logger.error(`[CLUSTER-STORE]: marking cluster as dead`, {
+              err,
+              name: clusterModel.contextName,
+              id: clusterModel.id,
+            });
+
+            notificationsStore.add({
+              message: `Cluster ${clusterModel.contextName} is reporting an error: ${err.toString()}`,
+              id: `${clusterModel.id}_IS_DEAD_NOTIFICATION`, // this **should** prevent double errors because we are loading the stores twice
+              status: NotificationStatus.ERROR,
+              timeout: 10000,
+            });
+            continue;
+          }
+
+          throw err;
```

Comment:
Are we catching this properly?

Reply:
Only if we were previously (which is to say, probably not). I catch the main error that happens now above.